thrift-analytics-publisher/src/main/java/org/wso2/am/thrift/analytics/publisher/ThriftClient.java

```diff
@@ -0,0 +1,121 @@
+import org.wso2.am.analytics.publisher.exception.MetricReportingException;
+import org.wso2.carbon.databridge.agent.DataPublisher;
+import org.wso2.carbon.databridge.agent.exception.DataEndpointAgentConfigurationException;
+import org.wso2.carbon.databridge.agent.exception.DataEndpointAuthenticationException;
+import org.wso2.carbon.databridge.agent.exception.DataEndpointConfigurationException;
+import org.wso2.carbon.databridge.agent.exception.DataEndpointException;
+import org.wso2.carbon.databridge.commons.Event;
+import org.wso2.carbon.databridge.commons.exception.TransportException;
+import org.wso2.carbon.databridge.commons.utils.DataBridgeCommonsUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Publishes analytics events to a Thrift server.
+ */
+public class ThriftClient {
+
+    private static final Logger log = LoggerFactory.getLogger(ThriftClient.class);
+    private static final String APIM_420_STREAM_VERSION = "4.2.0";
+    private final Gson gson;
+    private DataPublisher dataPublisher;
+
+    public ThriftClient(String type, String serverURL, String authURL, String serverUser,
+                        String serverPassword) {
+        try {
+            dataPublisher = createDataPublisher(type, serverURL, authURL, serverUser, serverPassword);
+        } catch (DataEndpointConfigurationException e) {
+            log.error("Error while creating the data publisher with configurations", e);
+        } catch (DataEndpointException | DataEndpointAgentConfigurationException | TransportException |
+                DataEndpointAuthenticationException e) {
+            log.error("Error while creating the data publisher", e);
+        }
+        gson = new Gson();
+    }
+
+    private DataPublisher createDataPublisher(String type, String serverURL, String authURL, String serverUser,
+                                              String serverPassword)
+            throws DataEndpointAuthenticationException, DataEndpointAgentConfigurationException, TransportException,
+            DataEndpointException, DataEndpointConfigurationException {
+        return new DataPublisher(type, serverURL, authURL, serverUser, serverPassword);
+    }
+
+    public void publish(ThriftMetricEventBuilder builder) throws MetricReportingException {
+        Map<String, Object> eventMap = builder.build();
+        ThriftStream thriftStream = builder.getThriftStream();
+        String streamName = thriftStream.getStreamName();
+
+        Object[] payload = generatePayload(thriftStream, eventMap);
+
+        Event event = new Event();
+        String streamId = DataBridgeCommonsUtils.generateStreamId(streamName, APIM_420_STREAM_VERSION);
+        event.setStreamId(streamId);
+        event.setMetaData(null);
+        event.setCorrelationData(null);
+        event.setPayloadData(payload);
+
+        try {
+//            printEventMap(eventMap); // Can be used to print the event for debugging purposes
```

Comment:
Let's remove this, and Improve and add some debug logs throughout the class

Reply:
Fixed via https://github.com/wso2/samples-apim/pull/119/commits/9ba77338d22d6c68238a1285e7ecfad7d0cd11c6